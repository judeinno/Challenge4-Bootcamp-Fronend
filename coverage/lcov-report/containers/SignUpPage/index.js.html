<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for containers/SignUpPage/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">containers/SignUpPage</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/12</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import * as PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { signupData, signupThunk } from '../../actions/signupActions';
import SignUp from '../../components/signup';
&nbsp;
class SignUpPage extends Component {
  handleChange = (event) =&gt; {
    event.preventDefault();
    const { signupUserData, signupReducer } = this.props;
    signupUserData({
      ...signupReducer.signupPostData,
      [event.target.name]: event.target.value,
    });
  };
&nbsp;
  handleSubmit = (event) =&gt; {
    event.preventDefault();
    const { signupUser, signupReducer } = this.props;
    signupUser(signupReducer.signupPostData);
  }
&nbsp;
  render() {
    return (
      &lt;SignUp
        onChange={this.handleChange}
        onClick={this.handleSubmit}
      /&gt;
    );
  }
}
&nbsp;
SignUpPage.propTypes = {
  signupUser: PropTypes.func.isRequired,
  signupUserData: PropTypes.func.isRequired,
  signupReducer: PropTypes.shape({}).isRequired,
};
&nbsp;
const mapStateToProps = ({ signupReducer }) =&gt; ({
  signupReducer,
});
&nbsp;
export const mapDispatchToProps = dispatch =&gt; ({
  signupUser: response =&gt; dispatch(signupThunk(response)),
  signupUserData: response =&gt; dispatch(signupData(response)),
});
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(SignUpPage);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 31 2018 12:42:11 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
