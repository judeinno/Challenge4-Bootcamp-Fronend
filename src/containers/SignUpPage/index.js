import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SignUp from '../../components/signup';

class SignUpPage extends Component {
  render() {
    const { textmessage } = this.props;
    return (
      <SignUp textmessage={textmessage} />
    );
  }
}

SignUpPage.propTypes = {
  textmessage: PropTypes.string.isRequired,
};

const mapStateToProps = ({ signupReducer }) => ({
  textmessage: signupReducer.textmessage,
});

export default connect(mapStateToProps)(SignUpPage);
