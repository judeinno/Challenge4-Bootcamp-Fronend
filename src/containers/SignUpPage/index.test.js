import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import SignUpPage, { mapDispatchToProps } from './index';

describe('<SignUpPage />', () => {
  let SignUpPageComponent;
  beforeEach(() => {
    jest.resetModules();
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    const initialState = { signupReducer: { signupPostData: { username: 'Jude', Password: 'password' } } };
    const store = mockStore(initialState);
    SignUpPageComponent = mount(
      <SignUpPage store={store} />,
    );
  });

  it('should render the component', () => {
    expect(SignUpPageComponent).toMatchSnapshot();
  });
  const dispatchTest = dispatch => (
    mapDispatchToProps(dispatch).signupUser({})
    && mapDispatchToProps(dispatch).signupUserData({})
  );
  it('should dispatch a method to get user input', () => {
    const dispatch = jest.fn();
    dispatchTest(dispatch);
  });

  it('should test user data without fail', () => {
    SignUpPageComponent
      .find('#user-name')
      .simulate('change', { target: { name: 'username', value: 'Jude' } });
    SignUpPageComponent
      .find('#sign-up-bt')
      .simulate('click');
  });
});
