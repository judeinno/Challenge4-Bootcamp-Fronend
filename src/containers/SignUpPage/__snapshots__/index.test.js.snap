// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage /> should render the component 1`] = `
<Connect(SignUpPage)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SignUpPage
    signupReducer={
      Object {
        "signupPostData": Object {
          "Password": "password",
          "username": "Jude",
        },
      }
    }
    signupUser={[Function]}
    signupUserData={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <SignUp
      onChange={[Function]}
      onClick={[Function]}
    >
      <div
        className="container signup"
      >
        <form
          className="signup-form"
        >
          <legend>
            Register
          </legend>
          <hr />
          <div
            className="form-group"
          >
            <p>
              Username:
            </p>
            <input
              className="form-control"
              id="user-name"
              name="username"
              onChange={[Function]}
              placeholder="User Name"
              type="test"
            />
          </div>
          <div
            className="form-group"
          >
            <p>
              Email:
            </p>
            <input
              aria-describedby="emailHelp"
              className="form-control"
              id="exampleInputEmail1"
              name="email"
              onChange={[Function]}
              placeholder="Enter email"
              type="email"
            />
          </div>
          <div
            className="form-group"
          >
            <p>
              Password:
            </p>
            <input
              className="form-control"
              id="exampleInputPassword1"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <button
            className="btn btn-primary"
            id="sign-up-bt"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </SignUp>
  </SignUpPage>
</Connect(SignUpPage)>
`;
